### PR TITLE
Directly use phpstan-wordpress extension

### LINKED ISSUE #4817: Directly use phpstan-wordpress extension
I've contributed static analysis to project when the WordPress extension was not published.
Now it is. And more advanced then ever.

Please consider using it.
https://packagist.org/packages/szepeviktor/phpstan-wordpress

Plus one: depends on php-stubs/wordpress-stubs so you don't have to depend on it

----

## Acceptance Criteria

Ping szepeviktor on the final PR for this

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Directly use phpstan-wordpress extension, removing the phpstan package.

Remove any copied over phpstan WordPress classes. 

Keep PHPStan extensions that work with service / container structure. 

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #4817

### CODE REVIEW

**Comment by spacedmonkey** [comment]
It is hard to test this working. I went in a broke and stuff and it detected that, but it seems to just work.

File: composer.json [hunk 1]
```diff
@@ -27,9 +27,9 @@
     "php-stubs/wordpress-stubs": "^v5.9.0",
```
Comment (swissspidy) [inline composer.json]: I think we can now remove this one as well

**Comment by swissspidy** [comment]
@szepeviktor as promised, pinging you here to let you know about this change that is finally happening :-)

We still include the `GetTermsDynamicFunctionReturnTypeExtension` class manually because there's no progress on https://github.com/szepeviktor/phpstan-wordpress/pull/50

**Review by swissspidy — APPROVED** [review]

**Comment by szepeviktor** [comment]
> because there's no progress on [szepeviktor/phpstan-wordpress#50](https://github.com/szepeviktor/phpstan-wordpress/pull/50)

You may finish that PR :)

**Comment by szepeviktor** [comment]
@swissspidy Please remove parts from the local config file already in my extension.
e.g. https://github.com/szepeviktor/phpstan-wordpress/blob/master/extension.neon#L68-L86

**Comment by szepeviktor** [comment]
@jeherve Please look at this PR how Pascal manages to run static analysis on this plugin.

**Comment by swissspidy** [comment]
> > because there's no progress on [szepeviktor/phpstan-wordpress#50](https://github.com/szepeviktor/phpstan-wordpress/pull/50)
> 
> You may finish that PR :)

I'll reach out to John and take a look 👍 

> @swissspidy Please remove parts from the local config file already in my extension. e.,g. [https://github.com/szepeviktor/phpstan-wordpress/blob/master/extension.neon#L68-L86](https://github.com/szepeviktor/phpstan-wordpress/blob/master/extension.neon?rgh-link-date=2022-02-03T08%3A30%3A11Z#L68-L86)

Ah yes, good catch!

**Pull request merged**